examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/favorites/blocs/favorite_puppies_bloc.dart

```diff
@@ -82,23 +86,16 @@ class FavoritePuppiesBloc
           puppy: updatedPuppy,
         ),
       );
-      errorDisplayed = false;
     } on Exception catch (e) {
-
-      final revertFavoritePuppy = puppy.copyWith(isFavorite: !isFavorite);
-
-      _coordinatorBloc.add(CoordinatorPuppyUpdatedEvent(revertFavoritePuppy));
-      if(errorDisplayed == false) {
-        yield state.copyWith(
-          favoritePuppies: state.favoritePuppies
-              .manageList(isFavorite: !isFavorite, puppy: puppy),
-          error: e.toString(),
-        );
-        errorDisplayed = true;
-      }
       yield state.copyWith(
         favoritePuppies: state.favoritePuppies
             .manageList(isFavorite: !isFavorite, puppy: puppy),
+        error: e.toString(),
+      );
+      await Future.delayed(const Duration(milliseconds: 200));
```

Comment:
@StanislavPrime, is there a real need of adding an artificial delay here?

Reply:
When I yield 2 different states from the bloc one after another it seems that the UI is not rebuilding when there is no delay and I cannot see the expected change of the Icon. I found a similar issue here:
https://github.com/felangel/bloc/issues/459
The explanation is: You're simply yielding states faster than flutter can rebuild the UI.

Reply:
I removed this delay from the method.